sdk/core/azure-core-amqp/src/amqp/message_receiver.cpp

```diff
@@ -257,9 +311,18 @@ namespace Azure { namespace Core { namespace Amqp { namespace _detail {
     {
       // LCOV_EXCL_START
       auto err = errno;
+#ifdef _MSC_VER
```

Comment:
Rather than pragma, why not switch to the same version?  If we can't then add comment for why we can't.

Reply:
@RickWinter that's the standard way to check if the code is being compiled using MSVVC compiler. It is needed so that MSVC-specific `#pragma warning` works under MSVC, but is being ignored for other compilers.

Reply:
What Anton said :).

This is a chunk of boilerplate to suppress the C4996 security warning.